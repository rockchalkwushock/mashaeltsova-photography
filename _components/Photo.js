import LazyLoad from 'react-lazyload'

import { createUrl } from '../lib'

const Photo = ({ id }) => (
  <LazyLoad height={200} once>
    <picture>
      <source media="(min-width: 1024px)" srcSet={createUrl(false, id, 320)} />
      <source media="(min-width: 768px)" srcSet={createUrl(false, id, 360)} />
      <source
        media="(min-width: 600px) and (orientation: landscape)"
        srcSet={createUrl(false, id, 275)}
      />
      <source media="(min-width: 540px)" srcSet={createUrl(false, id, 475)} />
      <source media="(min-width: 425px)" srcSet={createUrl(false, id, 375)} />
      <img src={createUrl(false, id, 300)} />
      <style jsx>{`
        img {
          -webkit-box-shadow: 6px 6px 6px rgba(54, 69, 79, 0.7);
          box-shadow: 6px 6px 6px rgba(54, 69, 79, 0.7);
          border-radius: 15px;
        }
      `}</style>
    </picture>
  </LazyLoad>
)

export default Photo
