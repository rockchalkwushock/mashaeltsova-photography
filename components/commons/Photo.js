import { Image, Transformation } from 'cloudinary-react'
import PropTypes from 'prop-types'

/**
 * REVIEW
 *
 * family - 8
 * portrait - 10
 * travel - 10
 * wedding - 8
 *
 */

const Photo = ({ publicId }) => (
  <div>
    <Image publicId={publicId} secure use_root_path>
      <Transformation
        crop="scale"
        dpr="auto"
        fetch_format="auto"
        quality="auto:best"
        width="320"
      />
    </Image>
    <style jsx>{`
      div {
        grid-column: span 8;
      }
      @media (min-width: 768px) {
        div {
          grid-column: span 12;
        }
      }
      @media (min-width: 1024px) {
        div {
          grid-column: span 12;
        }
      }
    `}</style>
  </div>
)

Photo.propTypes = {
  publicId: PropTypes.string.isRequired
}

export default Photo
