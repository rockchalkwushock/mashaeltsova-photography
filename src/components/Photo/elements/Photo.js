import React from 'react'
import Img from 'gatsby-image'
import styled from 'styled-components'

const UnstyledPhoto = props => <Img {...props} />

const Photo = styled(UnstyledPhoto)`
  border: none;
  box-shadow: 2px 2px 10px 1px rgba(0, 0, 0, 1);
  height: 240px;
  width: 240px;

  @media (min-width: ${({ theme }) => theme.sizes.sm}) {
    width: 400px;
  }
  @media (min-width: ${({ theme }) => theme.sizes.md}) {
    width: 300px;
  }
`
Photo.displayName = 'Photo'

export default Photo
