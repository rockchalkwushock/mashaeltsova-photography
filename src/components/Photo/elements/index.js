import React from 'react'
import Img from 'gatsby-image'
import styled from 'styled-components'

const UnstyledPhoto = props => <Img {...props} />

const StyledPhoto = styled(UnstyledPhoto)`
  border: none;
  box-shadow: 2px 2px 10px 1px rgba(0, 0, 0, 1);
`

StyledPhoto.displayName = 'StyledPhoto'

export default StyledPhoto
