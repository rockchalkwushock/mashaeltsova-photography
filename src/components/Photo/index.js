import React from 'react'

import { Photo } from './elements'

const AppPhoto = ({ alt, photo, title }) => (
  <Photo alt={alt} sizes={photo.sizes} title={title} />
)

export default AppPhoto
