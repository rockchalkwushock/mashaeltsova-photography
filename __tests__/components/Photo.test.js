import React from 'react'
import { render } from 'enzyme'

import Photo from '../../components/commons/photo/'

describe('Component: Photo', () => {
  test('Renders without exploding as site logo', () => {
    const tree = render(<Photo logo publicId="watermark" />)
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as a regular image', () => {
    const tree = render(
      <Photo onDismiss={() => {}} publicId="family_black_white" />
    )
    expect(tree).toMatchSnapshot()
  })
})
