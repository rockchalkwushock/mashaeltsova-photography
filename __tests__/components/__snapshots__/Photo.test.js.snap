// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Photo Renders without exploding as a regular image 1`] = `
<picture
  class="jsx-213633227 jsx-3672796762"
>
  <source
    class="jsx-213633227 jsx-3672796762"
    media="(min-width: 1024px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,q_auto:best,w_900/family_black_white"
  />
  <source
    class="jsx-213633227 jsx-3672796762"
    media="(min-width: 768px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,q_auto:best,w_650/family_black_white"
  />
  <source
    class="jsx-213633227 jsx-3672796762"
    media="(min-width: 600px) and (orientation: landscape)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,q_auto:best,w_575/family_black_white"
  />
  <source
    class="jsx-213633227 jsx-3672796762"
    media="(min-width: 540px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,q_auto:best,w_475/family_black_white"
  />
  <source
    class="jsx-213633227 jsx-3672796762"
    media="(min-width: 400px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,q_auto:best,w_375/family_black_white"
  />
  <img
    alt="family_black_white"
    class="jsx-213633227 jsx-3672796762 fadeIn"
    src="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,q_auto:best,w_300/family_black_white"
  />
</picture>
`;

exports[`Component: Photo Renders without exploding as site logo 1`] = `
<picture
  class="jsx-3672796762"
>
  <source
    class="jsx-3672796762"
    media="(min-width: 768px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,o_90,q_auto:best,w_575/watermark"
  />
  <source
    class="jsx-3672796762"
    media="(min-width: 540px) and (orientation: landscape)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,o_90,q_auto:best,w_300/watermark"
  />
  <source
    class="jsx-3672796762"
    media="(min-width: 540px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,o_90,q_auto:best,w_440/watermark"
  />
  <source
    class="jsx-3672796762"
    media="(min-width: 425px)"
    srcset="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,o_90,q_auto:best,w_375/watermark"
  />
  <img
    alt="Masha Eltsova Photography Logo"
    class="jsx-3672796762 fadeIn"
    src="https://res.cloudinary.com/mashaeltsovaphotography/c_scale,f_auto,o_90,q_auto:best,w_320/watermark"
  />
</picture>
`;
